Configuration/src/main/java/io/deephaven/configuration/Configuration.java

```diff
@@ -98,69 +105,83 @@ public static Configuration getNamed(@NotNull final String name) throws Configur
      */
     public static void reset() {
         DEFAULT = null;
         NAMED_CONFIGURATIONS.clear();
     }
 
     /**
-     * Clear the specified named configuration so it may be loaded anew.
-     * 
-     * @param name the configuration to clear.
+     * Clear the specified named Configuration so it may be loaded anew.
+     *
+     * @param name the Configuration to clear.
      */
-    public static void reset(final @NotNull String name) {
+    public static void reset(@NotNull final String name) {
         NAMED_CONFIGURATIONS.remove(name);
     }
 
-    public static Configuration newConfigurationForTesting() {
-        return newConfigurationForTesting(DEFAULT_CONF_NAME, DefaultConfigurationContext::new);
+    /**
+     * Create a new, non cached, default Configuration instance.
+     *
+     * @return a new Configuration instance, guaranteed to not be cached.
+     */
+    public static Configuration newStandaloneConfiguration() {
+        return newStandaloneConfiguration(DEFAULT_CONF_NAME, DefaultConfigurationContext::new);
     }
 
-    public static Configuration newConfigurationForTesting(final @NotNull String name) {
-        return newConfigurationForTesting(name, DefaultConfigurationContext::new);
+    /**
+     * Create a new, non cached, named Configuration instance.
+     *
+     * @param name the configuration name
+     * @return a new Configuration instance, guaranteed to not be cached.
+     */
+    public static Configuration newStandaloneConfiguration(@NotNull final String name) {
+        return newStandaloneConfiguration(name, DefaultConfigurationContext::new);
     }
 
+    /**
+     * Create a new, non cached, named Configuration instance.
+     *
+     * @param name the configuration name
+     * @param contextSupplier the supplier for {@link ConfigurationContext}
+     * @return a new Configuration instance, guaranteed to not be cached.
+     */
     @SuppressWarnings("UnusedReturnValue")
-    public static Configuration newConfigurationForTesting(final @NotNull String name,
-            final @NotNull Supplier<ConfigurationContext> contextSupplier) {
-        final Configuration newConfig = DEFAULT_CONF_NAME.equals(name)
+    public static Configuration newStandaloneConfiguration(@NotNull final String name,
+                                                           @NotNull final Supplier<ConfigurationContext> contextSupplier) {
+        return DEFAULT_CONF_NAME.equals(name)
                 ? new DefaultConfiguration(contextSupplier)
                 : new NamedConfiguration(name, contextSupplier);
-        init(newConfig);
-        return newConfig;
     }
 
-    protected Configuration(final @NotNull Supplier<ConfigurationContext> contextSupplier) {
+    protected Configuration(@NotNull final Supplier<ConfigurationContext> contextSupplier) {
         this.contextSupplier = contextSupplier;
     }
 
+    protected void init() {
```

Comment:
JavaDoc me? Do we anticipate users introducing their own implementations?

Reply:
No.  just another missed carryover from the first prototype.